icu4c/source/i18n/tibetancal.h

```diff
@@ -0,0 +1,313 @@
+
+#if !UCONFIG_NO_FORMATTING
+
+#include "unicode/calendar.h"
+
+U_NAMESPACE_BEGIN
+
+/**
+*  Concrete class which provides the Tibetan calendar.
+*
+*
+*
+*
+*
+*
+*
+*/
+
+
+class U_I18N_API TibetanCalendar : public Calendar {
+public:
+  /**
+   * Useful constants for TibetanCalendar.
+   * @internal
+   */
+
+
+  /**
+   * Calendar type - phugpa or tsurphu
+   * @internal 
+   */
+  enum ECalculationType {
+    PHUGPA,
+    TSURPHU
+  };
+
+  //-------------------------------------------------------------------------
+  // Constructors...
+  //-------------------------------------------------------------------------
+
+  /**
+   * Constructs an TibetanCalendar based on the current time in the default time zone
+   * with the given locale.
+   *
+   * @param aLocale  The given locale.
+   * @param success  Indicates the status of TibetanCalendar object construction.
+   *                 Returns U_ZERO_ERROR if constructed successfully.
+   * @param type     The Tibetan calendar calculation type. The default vaue is PHUGPA.
+   * @internal
+   */
+  TibetanCalendar(const Locale& aLocale, UErrorCode &success, ECalculationType type = PHUGPA);
+
+
+  /**
+   * Copy Constructor
+   * @internal
+   */
+  TibetanCalendar(const TibetanCalendar& other);
+
+
+  /**
+   * Destructor.
+   * @internal
+   */
+  virtual ~TibetanCalendar();
+
+
+  /**
+   * Sets Tibetan calendar calculation type used by this instance.
+   *
+   * @param type    The calendar calculation type, <code>PHUGPA</code> to use the phugpa
+   *                calendar, <code>TSURPHU</code> to use the tsurphu calendar.
+   * @internal
+   */
+  void setCalculationType(ECalculationType type, UErrorCode &status);
+
+
+  /**
+   * Returns <code>true</code> if this object is using the phugpa
+   * calendar, or <code>false</code> if using the tsurphu calendar.
+   * @internal
+   */
+  UBool isPhugpa();
+
+
+  // clone
+  virtual TibetanCalendar* clone() const;
+
+
+
+  /**
+   * return the calendar type, "tibetan-phugpa" or "tibetan-tsurphu".
+   *
+   * @return calendar type
+   * @internal
+   */
+  virtual const char * getType() const;
+
+
+private:
+
+  //-------------------------------------------------------------------------
+  // Internal data....
+  //-------------------------------------------------------------------------
+
+/**
+ * Return the moonTab by using the table for the given integer value. 
+ * @param moonTab An integer number
+ */
+int32_t moonTab(int32_t moonTab) const;
+
+
+/**
+ * Return the sunTab by using the table for the given integer value. 
+ * @param sunTab An integer number
+ */
+int32_t sunTab(int32_t sunTab) const;
+
+
+/**
+ * Return the modulo of the number if(num % mod != 0) else is return mod.
+ * @param num the number to be divided
+ * @param mod the number to be divided with
+ */
+int32_t amod(int32_t num, int32_t mod) const;
+
+
+/**
+ * Return the month count(based on the epoch) from the given Tibetan year, month number and leap month indicator.
```

Comment:
nit: Return -> Returns
I assume the "mount count" means the ordinal month of that year. Please indicate is it 0-base or 1-base (is the first month in that year return 0 or 1) in the header file. Also, make this an protected virtual function and override it in a protected function in TibetanTsurphuCalendar.

Reply:
What is the possible value of is_leap_month ? I assume it is a bool.
How about rename it to virtual int32_t ordinalMonth(int32_t eyear, int32_t month, bool isLeapMonth) const;